File: gcc/testsuite/gfortran.dg/gomp/pr104757.f90
! PR middle-end/104757
! { dg-do compile }
! { dg-options "-O -fopenmp" }

module pr104757
  implicit none (external, type)
  integer :: ll
  !$omp declare target (ll)
contains
  subroutine foo (i1)
    !$omp declare target (foo)
    logical :: i1
    integer :: i
    !$omp distribute simd if(i1)
    do i = 1, 64
      ll = ll + 1
    end do
  end
end module

File: gcc/testsuite/gcc.dg/gomp/pr104757.c
/* PR middle-end/104757 */
/* { dg-do compile } */
/* { dg-options "-O2 -fopenmp" } */
// { dg-additional-options "-Wno-deprecated-openmp" }
#pragma omp declare target
void
foo (int x, int y, int *z)
{
  int j = 0;
  #pragma omp simd linear(j:x + y)
  for (int i = 0; i < 64; i++)
    j += x + y;
}
#pragma omp end declare target
